autoscaler/azure.py

```diff
@@ -58,6 +60,51 @@ def add_done_callback(self, fn):
                 self._callbacks.append(fn)
 
 
+# XXX: Azure sometimes sends us a Retry-After: 1200, even when we still have quota, causing our client to appear to hang.
+# Ignore them and just retry after 30secs
+_RETRY_TIME_LIMIT = 30
+
+
+class AzureRetry(ClientRetry):
```

Comment:
It's quite Azure specific (see the logging code), but I'll make it something a bit more descriptive